Update mutate_pdb.py to use protein sequence/pdb numbering rather than pose numbering

mglubber opened the pull request:
Update mutate_pdb.py to use protein sequence/pdb numbering rather than pose numbering. Added a function converts the numbering sequence from the pdb/protein sequence (which is consistent across papers and pdb files) to the pose numbering (which varies with the pose) automatically. Allows mutations to be specified using notation directly from papers such as Kleinstivee et al. 2015.

mglubber committed:
mutate_pdb.py - added function to convert protein sequence/pdb file aa numbering into PyRosetta internal pose number. This makes specifying aa mutations significantly easier
mutate_to_VQR_EQR.py has similar additions, plus comments on variations to try for packing.

mglubber committed:
mutate_pdb.py - added function to convert protein sequence/pdb file a numbering into PyRosetta internal pose number. This makes specifying aa mutations significantly easier mutate_to_VQR_EQR.py has similar additions, plus comments on variations to try for packing.

mattsmart commented on models/tridimensional/docking_validation/mutate_pdb.py:
```diff
@@ -61,6 +64,23 @@ def mutate_pdb(input_pdb_path, mutations, output_directory, output_id):
+
+def findPyRosettaResNum(pose,chain,pdb_res_num):
+    ''' Find the internal PyRosetta number given the protein sequence numbering.
+        Args:
```
move args back a tab

mattsmart commented on models/tridimensional/docking_validation/mutate_pdb.py:
```diff
@@ -61,6 +64,23 @@ def mutate_pdb(input_pdb_path, mutations, output_directory, output_id):
+    pose_res_num = pose.pdb_info().pdb2pose(chain,pdb_res_num)
+    assert pose_res_num != 0; "Amino acid number %r is not a valid position in the pose" % pdb_res_num
+    return pose_res_num
     
```
extra newline

mattsmart commented on models/tridimensional/docking_validation/mutate_pdb.py:
```diff
@@ -61,6 +64,23 @@ def mutate_pdb(input_pdb_path, mutations, output_directory, output_id):
+    DNA and sgRNA chains present.
+    '''
+    pose_res_num = pose.pdb_info().pdb2pose(chain,pdb_res_num)
+    assert pose_res_num != 0; "Amino acid number %r is not a valid position in the pose" % pdb_res_num
```
i think you want a comma instead of a semicolon here

also, why `%r` instead of `%d`? is `pdb_res_num` not an int?

mattsmart commented on models/tridimensional/docking_validation/mutate_pdb.py:
```diff
@@ -61,6 +64,23 @@ def mutate_pdb(input_pdb_path, mutations, output_directory, output_id):
     return output_pdb_path
 
+
+def findPyRosettaResNum(pose,chain,pdb_res_num):
```
find_pyrosetta_res_num

mattsmart commented on models/tridimensional/docking_validation/mutate_pdb.py:
```diff
@@ -33,7 +35,8 @@ def mutate_pose(pose, mutations):
     # Let's release the PI domain
     for i in range(1110, 1388):
-        pose_packer.temporarily_set_pack_residue(i, True)
+        pose_num = findPyRosettaResNum(mutate_pose, 'B', aa_num)
```
what is this line doing? its not using the iterant `i` (comparing with line 25)

mattsmart commented on models/tridimensional/docking_validation/mutate_pdb.py:
```diff
@@ -33,7 +35,8 @@ def mutate_pose(pose, mutations):
     for i in range(1110, 1388):
-        pose_packer.temporarily_set_pack_residue(i, True)
+        pose_num = findPyRosettaResNum(mutate_pose, 'B', aa_num)
+        pose_packer.temporarily_set_pack_residue(pose_num, True)
```
this line also isn't using the loop variable `i`?

mattsmart commented on models/tridimensional/incomplete_scripts/mutate_to_VQR_EQR.py:
```diff
@@ -52,6 +59,7 @@ def findPyRosettaResNum(pose,chain,pdb_res_num):
     DNA and sgRNA chains present.
     '''
     pose_res_num = pose.pdb_info().pdb2pose(chain,pdb_res_num)
+    assert pose_res_num != 0; "Amino acid residue number %r is not a valid position in the pose." % pdb_res_num
```
semicolon-> comma i think

mglubber committed:
Updated according to Matt's suggestion. Corrected issue with aa_num instead of loop variable being use when freezing PI domain, plus formatting issues.

mglubber committed:
Some more formatting because source tree is slow

mglubber committed:
generate_pams_chimera.py does not import required function pam_string_from_int from utility.py - this breaks the script. Added import function and tested.
Also created validation_test version of generate_pams_chimera to test if replacing existing pam with identical sequence results in differences in the scoring of the DNA interface. This will confirm if the current scoring is solely due to nucleotides, or if pam generation method has an effect

mattsmart commented on models/tridimensional/docking_validation/mutate_pdb.py:
```diff
@@ -61,6 +64,25 @@ def mutate_pdb(input_pdb_path, mutations, output_directory, output_id):
+
 if __name__ == '__main__':
     print "main behaviour not yet implemented"
+ 
```
you killed the EOF newline with latest commit :(

mattsmart commented on models/tridimensional/docking_validation/generate_pams_chimera.py:
```diff
@@ -8,6 +8,7 @@
 from chimera import runCommand
 from chimera import replyobj
 from constants import PAM_TEMPLATE_SEQUENCE, DNA_ALPHABET
+from utility import pam_string_from_int 
```
good catch

mglubber committed:
EOF newline for matt. Because apparently he feels lonely without it.

mattsmart commented:
once you get rid of that CaMeLcase heresy in your new function name it LGTM

mglubber committed:
CaMeLcase = bestcase

mattsmart commented:
LGTM
